mtags/src/main/scala/scala/meta/internal/mtags/MtagsEnrichments.scala

```diff
@@ -311,11 +311,18 @@ trait MtagsEnrichments {
         range.getEnd.getCharacter
       )
     def encloses(position: l.Position): Boolean = {
```

Comment:
Unless I'm misunderstanding how ranges work, I think this logic was incorrect for multi-line ranges (e.g. a range from char 10 of line 10 to char 20 of line 12 should enclose the position at char 5 of line 11) so I rewrote it. 

I noticed the same mistake in a few other `encloses` methods in this file and in `MetalsEnrichment.scala`. I can fix those too, but probably better to do that in a separate PR.